### PR TITLE
[PanelStack] documentation

### PR DESCRIPTION
**Pull request by giladgray**
- major edits to `PanelStack` docs

cc @kadhirvelm

### CODE REVIEW

File: packages/core/src/components/panel-stack/panel-stack.md [hunk 1]
```diff
@@ -1,25 +1,55 @@
 @# Panel stack
 
-`PanelStack` is a component designed to render multiple panels from a stack, but only display the
-top most panel at any given time.
+`PanelStack` manages a stack of panels and displays only the topmost panel.
 
-It will render each panel with a header and uses [`CSSTransition`](http://reactcommunity.org/react-transition-group/css-transition)
-to seamlessly move between panels. Whenever there is more than a single panel on the stack, `Panel Stack` will display a back
-button allowing the user to easily move to the previous panel.
+Each panel appears with a header containing a "back" button to pop the stack and
```
Comment (kadhirvelm) [inline packages/core/src/components/panel-stack/panel-stack.md]: "...to pop **off** the stack..."

File: packages/core/src/components/panel-stack/panel-stack.md [hunk 2]
```diff
@@ -1,25 +1,55 @@
...
+
+<PanelStack initialPanel={{ component: MyPanel, title: "Home" }} />
+```
+
+@interface IPanelProps
+
 @## Props
 
+Panels are supplied as `{ component, props, title }` objects where `component`
+and `props` are used to render the panel element and `title` will appear in the
```
Comment (kadhirvelm) [inline packages/core/src/components/panel-stack/panel-stack.md]: We should match the tense in the second clause: "...panel element while `title` is used in the header and to label the back button."

File: packages/core/src/components/panel-stack/panelProps.ts [hunk 3]
```diff
@@ -23,40 +23,33 @@ export interface IPanel<P = {}> extends IPanelOptions {
...
     closePanel(): void;
 
     /**
-     * Call this method to open a new panel on the top of the stack.
-     * @param component The React component of the new panel.
-     * @param props The props to be passed to the new panel.
-     * @param options Additional options for the new panel.
+     * Call this method to open a new panel on the top of the stack. Together,
+     * `component` and `props` are used to render the panel element when needed
+     * (and `IPanelProps` is injected). The `title` option is required and will
```
Comment (kadhirvelm) [inline packages/core/src/components/panel-stack/panelProps.ts]: "...element when needed – **`IPanelProps` will be injected on render.** The `title`..."

Comment (giladgray) [reply packages/core/src/components/panel-stack/panelProps.ts]: 😕 not great use of the dash. it really is a side thought, perfect for a parenthetical.

Comment (giladgray) [reply packages/core/src/components/panel-stack/panelProps.ts]: but actually #2738 removes all these docs 👍

File: packages/docs-theme/src/styles/_props.scss [hunk 4]
```diff
@@ -123,6 +123,6 @@
 
 // modifiers table parent
 .docs-modifiers {
-  margin-top: $heading-margin;
-  margin-bottom: $heading-margin / 2;
+  margin-top: $heading-margin / 2;
```
Comment (kadhirvelm) [inline packages/docs-theme/src/styles/_props.scss]: Is this supposed to be here?

Comment (giladgray) [reply packages/docs-theme/src/styles/_props.scss]: yeah i did that :)

**Pull request merged**